activerecord/lib/active_record/relation/calculations.rb

```diff
@@ -167,7 +183,7 @@ def pluck(*column_names)
       else
         relation = spawn
         relation.select_values = column_names.map { |cn|
-          @klass.has_attribute?(cn) || @klass.attribute_alias?(cn) ? arel_attribute(cn) : cn
+          @klass.effective_column_name?(cn) ? arel_attribute(cn) : cn
         }
         result = klass.connection.select_all(relation.arel, nil, bound_attributes)
         result.cast_values(klass.attribute_types)
```

Comment:
… However, having separate API for `foo` vs `unsafe_foo` is clearer. Alternatively, we could share a private pluck impl method.